src/Core/Resources/Resources.cpp

```diff
@@ -55,6 +56,29 @@ optional<std::string> getResourcesPath( void* symbol, const std::string& pattern
     return p.string();
 }
 
+namespace DataPath {
+static std::stack<std::string> s_dataPaths;
+}
+
+std::string getDataPath() {
+    if ( DataPath::s_dataPaths.empty() )
+    { DataPath::s_dataPaths.emplace( fs::current_path().string() ); }
```

Comment:
Do that mean that if one call pop before get, there is a core assert, and if pop is called after get, everything is fine? 
```suggestion
    { return fs::current_path().string(); }
```

Reply:
Yes, calling get never generates an error, even if the stack is empty. In this case (first call to get), the default path is pushed and get return this default pushed path and the stack is not empty after that. Your suggestion is also fine. If the stack is empty, return the default.
We can also change the pop method so that if the stack is empty, nothing happen.

Reply:
both behavior are fine for me. I prefer not to have the current version with side effect.

Reply:
OK, I'll change that